Tutorial notebook for coadd catalog with GCR access

EiffL opened the pull request:
I'm opening this PR to pin this here, and have a place for discussion.

This is addressing #185 I think in good shape for early alpha review, but still has a few placeholders. Here is the link to the [notebook](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20access%20--%20Tutorial%20(part1).ipynb)

wmwv commented on Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb:
```diff
@@ -0,0 +1,1401 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mask = np.zeros_like(m5map)\n",
```
`m5map` has not yet been defined in this Notebook.

wmwv commented on Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb:
```diff
@@ -0,0 +1,1401 @@
+    "hist2d(data_cut['ra'], data_cut['dec'],256); gca().set_aspect('equal');\n",
+    "xlabel('RA [deg]');\n",
+    "ylabel('dec [deg]');\n",
+    "title('Clean objects');"
```
Add a colorbar.

wmwv commented on Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb:
```diff
@@ -0,0 +1,1401 @@
+     "output_type": "execute_result"
+    },
+    {
+     "data": {
```
Please remove plots.  If you have an image you want to persist, make a PNG, and include in assets.

wmwv commented on Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb:
```diff
@@ -0,0 +1,1401 @@
+    },
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAoIAAAFBCAYAAADjZM6SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAIABJREFUeJzsvX1wHdd55vkcAAQIECLBT5CgRFMUJdmkY+vDlixnNptEdiJ5dyw7u9HaNWUrHx5Zu/Y4ySZVkRNVTbbWU1KlKklZGa8dDeOKrJ2JrSiTMTdR5LXlTCytpUgWbSumNLRBiqIIiJ8iQJEAAQI4+0eft8/bt9/bfXFxAd6L+/yqVGyePuf06b73tprv08/7Ou89CCGEEEJI+9FxqRdACCGEEEIuDXwQJIQQQghpU/ggSAghhBDSpvBBkBBCCCGkTeGDICGEEEJIm8IHQUI ...
```
Please remove plots.  If you have an image you want to persist, make a PNG, and include in assets.

wmwv replied:
Remove output

wmwv replied:
Remove output.

wmwv commented on Notebooks/DC2 Coadd Run1.1p GCR access -- Tutorial (part1).ipynb:
```diff
@@ -0,0 +1,1401 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pylab inline"
```
I'm very much a fan of being explicitly about `pyplot` and `numpy` use.  These tutorials also will end up being examples that people will copy for style.

I would suggest changing this one line to:
```
%matplotlib inline

import matplotlib.pyplot as plt
import numpy as np
```

and then encourage use of `np.` and `plt.` as neeeed below.

EiffL commented:
@wmwv we have split the notebooks in 3, and I have renamed them as:
- [DC2 Coadd Run1.1p GCR tutorial -- Part I: GCR Access.ipynb](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20I:%20GCR%20Access.ipynb)
- [DC2 Coadd Run1.1p GCR tutorial -- Part II: Lensing Cuts.ipynb](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20II:%20Lensing%20Cuts.ipynb)
- [DC2 Coadd Run1.1p GCR tutorial -- Part III: Guided Challenges](https://github.com/LSSTDESC/DC2_Repo/blob/u/EiffL/cmu_tutorial/Notebooks/DC2%20Coadd%20Run1.1p%20GCR%20tutorial%20--%20Part%20III:%20Guided%20Challenges.ipynb)

yymao commented:
@EiffL how about updating the format of `native_filters` to `['tract==<tract_number>']` (lambda function still works, but string is easier to read).

EiffL commented:
Sure, I'll do that and remove the last comments/placeholders we have

wmwv commented:
Can you rename to not have -- or : in the filename, and potentially remove the spaces (I'm planning to go through later and rename the rest to not have spaces)?
: is an alternate path separator on Mac OSX.
-- is the "process no more options" indicator.

While obviously these all should be escaped, just like the spaces should be, in practice I think we will be better off to omit them.

You can still have them in the title of the Notebook, of course.

EiffL commented:
Ok, thanks @wmwv I think notebooks I and II are in very good shape now, @yymao I've updated the code to use the new `native_filters` syntax, thanks for implementing that!

EiffL commented:
Concerning Part III, which we intended as guided challenges, with incomplete cells that people would fill-in, how do you feel about them?

EiffL commented:
Ok, rebased to master :-)
